sources/packages/backend/apps/api/src/route-controllers/application-restriction-bypass/models/application-restriction-bypass.dto.ts

```diff
@@ -15,3 +21,41 @@ export class ApplicationRestrictionBypassSummary {
 export class ApplicationRestrictionBypassHistoryAPIOutDTO {
   bypasses: ApplicationRestrictionBypassSummary[];
 }
+
+export class BypassRestrictionAPIInDTO {
+  @IsPositive()
+  applicationId: number;
+  @IsPositive()
+  studentRestrictionId: number;
+  @IsEnum(RestrictionBypassBehaviors)
+  bypassBehavior: RestrictionBypassBehaviors;
+  @IsString()
+  @MaxLength(NOTE_DESCRIPTION_MAX_LENGTH)
+  note: string;
+}
+
+export class RemoveBypassRestrictionAPIInDTO {
+  @IsString()
+  @MaxLength(NOTE_DESCRIPTION_MAX_LENGTH)
+  note: string;
+}
+
+export class AvailableStudentRestrictionAPIOutDTO {
+  availableRestrictionsToBypass: {
+    studentRestrictionId: number;
+    restrictionCode: string;
+    studentRestrictionCreatedAt: Date;
+  }[];
+}
+export class ApplicationRestrictionBypassAPIOutDTO {
+  applicationRestrictionBypassId: number;
+  studentRestrictionId: number;
+  restrictionCode: string;
+  applicationRestrictionBypassCreatedDate: string;
+  applicationRestrictionBypassCreatedBy: string;
+  applicationRestrictionBypassCreationNote: string;
+  applicationRestrictionBypassRemovedDate?: string;
+  applicationRestrictionBypassRemovedBy?: string;
+  applicationRestrictionBypassRemovalNote?: string;
+  applicationRestrictionBypassBehavior: string;
```

Comment:
In a DTO already named `ApplicationRestrictionBypass` I would suggest removing the prefix `applicationRestrictionBypass` from all these properties.